packages/core/epoch/commitmentmanager/commitmentmanager.go

```diff
@@ -0,0 +1,129 @@
+package commitmentmanager
+
+import (
+	"sync"
+
+	"github.com/cockroachdb/errors"
+	"github.com/iotaledger/hive.go/core/generics/walker"
+
+	"github.com/iotaledger/goshimmer/packages/core/epoch"
+)
+
+type CommitmentManager struct {
+	Events             *Events
+	SnapshotCommitment *Commitment
+
+	commitmentsByEC map[epoch.EC]*Commitment
```

Comment:
I found it a little confusing at first - "commitments by Epoch Commitment" sounds a little weird imo. Should we denote somehow when we use hash (merkle root) and when the contents of a commitment?